sdk/core/System.Memory.Data/src/BinaryData.cs

```diff
@@ -52,7 +52,7 @@ public BinaryData(byte[] data)
         public BinaryData(object? jsonSerializable, JsonSerializerOptions? options = default, Type? type = default)
 #pragma warning restore AZC0014 // Avoid using banned types in public API
         {
-            _bytes = JsonSerializer.SerializeToUtf8Bytes(jsonSerializable, type, options);
+            _bytes = JsonSerializer.SerializeToUtf8Bytes(jsonSerializable, type ?? jsonSerializable?.GetType() ?? typeof(object), options);
```

Comment:
Do we want typeof(object) or jsonSerializable.GetType()? I think we want the later. typeof(object) is a pretty useless payload to serialize.

Reply:
Yes, we first try jsonSerializable.GetType:
``` 
type ?? jsonSerializable?.GetType() ?? typeof(object)
```

Reply:
Just to clarify - are you suggesting that we don't fall back to typeof(object) at all?

Reply:
Yeah, I would not fall back to typeof(object).

Reply:
Also, as of .NET 5.0, null is not allowed to be passed here - https://github.com/dotnet/runtime/pull/44633#discussion_r522693231

Reply:
Right, so today when passing in `jsonSerializable: null, type: null` you get back `null` from the serializer.  In 5.0 this [throws](https://docs.microsoft.com/en-us/dotnet/core/compatibility/serialization#jsonserializerserialize-throws-argumentnullexception-when-type-parameter-is-null) ANE.  By falling back to passing `typeof(object)` here you'll have consistent behavior which is `null`.